src/node_mutex.h

```diff
@@ -15,6 +17,50 @@ struct LibuvMutexTraits;
 using ConditionVariable = ConditionVariableBase<LibuvMutexTraits>;
 using Mutex = MutexBase<LibuvMutexTraits>;
 
+template <typename T, typename MutexT = Mutex>
+class ExclusiveAccess {
+ public:
+  ExclusiveAccess() = default;
+
+  template <typename... Args>
+  explicit ExclusiveAccess(Args... args) : item_(args...) {}
```

Comment:
```suggestion
  explicit ExclusiveAccess(Args&&... args) : item_(std::forward<Args>(args)...) {}
```

to pass on references correctly instead of implicitly copying arguments?

Reply:
Done 👍